scr/main/java/bittercode/model/Book.java

```diff
@@ -0,0 +1,65 @@
+package com.bittercode.model;
+
+import java.io.Serializable;
+
+public class Book implements Serializable {
+
+    private String barcode;
+    private String name;
+    private String author;
+    private double price;
+    private int quantity;
+
+    public Book(String barcode, String name, String author, double price, int quantity) {
+        this.barcode = barcode;
+        this.name = name;
+        this.author = author;
+        this.setPrice(price);
+        this.quantity = quantity;
+    }
+
+    public Book() {
+        super();
+    }
+
+    public String getBarcode() {
+        return barcode;
+    }
+
+    public void setBarcode(String barcode) {
+        this.barcode = barcode;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public String getAuthor() {
+        return author;
+    }
+
+    public void setAuthor(String author) {
+        this.author = author;
+    }
+
+    public int getQuantity() {
+        return quantity;
+    }
+
+    public void setQuantity(int quantity) {
+        this.quantity = quantity;
+    }
+
+    public double getPrice() {
+        return price;
+    }
+
+    public void setPrice(double price) {
+        this.price = price;
+    }
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider price validation in setter method</b></div>
<div id="fix">

Consider adding validation in `setPrice()` to ensure price is not negative. This would help prevent invalid data states. A similar issue was also found in scr/main/java/bittercode/model/Book.java (line 53-55).
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -61,3 +61,6 @@
     public void setPrice(double price) {
 +        if (price < 0) {
 +            throw new IllegalArgumentException("Price cannot be negative");
 +        }
         this.price = price;
     }
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/PrajaktaBendreBito/BookStore/pull/1#issuecomment-2514333017>#855223</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged